internal/contentmanager/contentmanager.go

```diff
@@ -0,0 +1,203 @@
+package contentmanager
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/operator-framework/operator-controller/api/v1alpha1"
+	oclabels "github.com/operator-framework/operator-controller/internal/labels"
+)
+
+type ContentManager interface {
```

Comment:
Typically I wouldn't define the interface where it is implemented, but we wanted to have a good starting point without having to wire everything together (scoped for #975) and created this interface based on what we think would be needed by the ClusterExtensionReconciler.

This interface and code will likely get moved around a bit when we wire everything up in #975

Reply:
I was about to note that it hasn't been wired up.

Reply:
> Typically I wouldn't define the interface where it is implemented

I know this was on advice from Andy, but I haven't gotten used to this pattern at all and I think we may be taking it to an extreme. Looking at the Go standard library, locations of interfaces are fairly intuitive (e.g. `io.Reader` lives in the `io` package where there are lots of methods that accept or return it. `json.Marshaler` lives in the `encoding/json` package, where JSON marshaling/unmarshaling implementations exist.

To me, it is super-unituitive for our `controllers` package to be the place where the InstalledBundleGetter and Preflight interfaces are defined. https://github.com/operator-framework/operator-controller/blob/8947c3de2799b7bd765a4a586b81fd82ea668e2e/internal/controllers/clusterextension_controller.go#L103-L120

Reply:
I think it is more so that the interfaces are defined where they are _used_. That being said, this isn't a hill I'm willing to die on, but it is a pattern I've come to appreciate.